content/en-US/api/URLPattern.md

```diff
@@ -0,0 +1,515 @@
+
+// `cdn.example.com`
+result.hostname.input;
+
+// [{ hostname: "cdn.example.com" }]
+result.inputs;
+```
+
+### Example: Accessing matched group values using custom names.
+
+The following example shows how groups can be given custom names which can
+be used to accessed the matched value in the result object.
+
+```js
+// Construct a URLPattern using matching groups with custom names.  These
+// names can then be later used to access the matched values in the result
+// object.
+const pattern = new URLPattern({ pathname: "/:product/:user/:action" });
+const result = pattern.exec({ pathname: "/store/wanderview/view" });
+
+// `store`
+result.pathname.groups.product;
+
+// `wanderview`
+result.pathname.groups.user;
+
+// `view`
+result.pathname.groups.action;
+
+// `/store/wanderview/view`
+result.pathname.input;
+
+// [{ pathname: "/store/wanderview/view" }]
+result.inputs;
+```
+
+### Example: Custom regular expression groups.
+
+The following example shows how a matching group can use a custom regular
+expression.
+
+```js
+const pattern = new URLPattern({ pathname: "/(foo|bar)" });
+
+// true
+pattern.test({ pathname: "/foo" });
+
+// true
+pattern.test({ pathname: "/bar" });
+
+// false
+pattern.test({ pathname: "/baz" });
+
+const result = pattern.exec({ pathname: "/foo" });
+
+// `foo`
+result.pathname.groups[0];
+```
+
+### Example: Named group with a custom regular expression.
+
+The following example shows how a custom regular expression can be used with a
+named group.
+
+```js
+const pattern = new URLPattern({ pathname: "/:type(foo|bar)" });
+const result = pattern.exec({ pathname: "/foo" });
+
+// `foo`
+result.pathname.groups.type;
+```
+
+### Example: Making matching groups optional.
+
+The following example shows how a matching group can be made optional by placing
+a `?` modifier after it.  For the pathname component this also causes any
+preceding `/` character to also be treated as an optional prefix to the group.
+
+```js
+const pattern = new URLPattern({ pathname: "/product/(index.html)?" });
+
+// true
+pattern.test({ pathname: "/product/index.html" });
+
+// true
+pattern.test({ pathname: "/product" });
+
+const pattern2 = new URLPattern({ pathname: "/product/:action?" });
+
+// true
+pattern2.test({ pathname: "/product/view" });
+
+// true
+pattern2.test({ pathname: "/product" });
+
+// Wildcards can be made optional as well.  This may not seem to make sense
+// since they already match the empty string, but it also makes the prefix
+// `/` optional in a pathname pattern.
+const pattern3 = new URLPattern({ pathname: "/product/*?" });
+
+// true
+pattern3.test({ pathname: "/product/wanderview/view" });
+
+// true
+pattern3.test({ pathname: "/product" });
+
+// true
+pattern3.test({ pathname: "/product/" });
+```
+
+### Example: Making matching groups repeated.
+
+The following example shows how a matching group can be made repeated by placing
+a `+` modifier after it.  Again, in the pathname component this also treats the
```

Comment:
```suggestion
a `+` modifier after it.  In the pathname component this also treats the
```

Reply:
This is honestly a bit of a judgement call, but here's the reasoning. In a reference work, readers aren't necessarily reading sequentially. I don't like to imply that they need to. The document as a whole still works sequentially if someone choses to read it that way.